libs/community/langchain_community/graph_vectorstores/extractors/link_extractor_transformer.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Iterable, Sequence
+
+from langchain_core.documents import Document
+from langchain_core.documents.transformers import BaseDocumentTransformer
+from langchain_core.graph_vectorstores.links import add_links
+
+from langchain_community.graph_vectorstores.extractors.link_extractor import (
+    LinkExtractor,
+)
+
+
+class LinkExtractorTransformer(BaseDocumentTransformer):
+    """DocumentTransformer for applying one or more LinkExtractors.
+
+    Example:
+        .. code-block:: python
+
+            extract_links = LinkExtractorTransformer([
+                HtmlLinkExtractor().as_document_extractor(),
+            ])
+            extract_links.transform_documents(docs)
+    """
+
+    def __init__(self, link_extractors: Iterable[LinkExtractor[Document]]):
+        """Create a DocumentTransformer which adds extracted links to each document."""
+        self.link_extractors = link_extractors
+
+    def transform_documents(
+        self, documents: Sequence[Document], **kwargs: Any
+    ) -> Sequence[Document]:
+        # Implement `transform_docments` directly, so that LinkExtractors which operate
+        # better in batch (`extract_many`) get a chance to do so.
+        document_links = zip(
```

Comment:
Do you have examples where `extract_many` is useful in terms of performance or is this done in anticipation?

nit: There should be a way to rephrase this code to be easier to read without sacrificing performance. At least I find the double usage of `zip` and `*` to be hard to follow.

Reply:
Yeah -- I'll send some PRs in the near future, but things like Keybert use a model to extract keywords. They are able to batch the application of the model across many documents, so it is much more efficient.

https://colab.research.google.com/drive/174MxzfqwQqHeOIuGTW0_E6WFGvbT3krf#scrollTo=BLdNC65TCju6

On 100 random strings of length 8:

```
Map:   3.4769s
Batch:  0.9578s
```